graylog2-server/src/main/java/org/graylog2/indexer/Deflector.java

```diff
@@ -297,4 +298,22 @@ public boolean isGraylogDeflectorIndex(final String indexName) {
     public boolean isGraylogIndex(final String indexName) {
         return !isNullOrEmpty(indexName) && !isDeflectorAlias(indexName) && indexPattern.matcher(indexName).matches();
     }
+
+    /**
+     * Sorts the given set of index names and removes the deflector alias from all indices but the last one.
+     *
+     * This might be used when multiple indices have the same alias. This should be really rare!
+     *
+     * @param indexNames the set of index names that have the deflector alias
+     */
+    public void cleanupAliases(Set<String> indexNames) {
+        final Set<String> indicesToRemove = indexNames.stream()
+                .mapToInt(Deflector::extractIndexNumber)
+                .sorted()
+                .limit(indexNames.size() - 1) // Get all but the newest one
+                .mapToObj(num -> indexPrefix + SEPARATOR + num)
```

Comment:
Hm, in the worst case it will already fail in line 311 with a `NumberFormatException`.